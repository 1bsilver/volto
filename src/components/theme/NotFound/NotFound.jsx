/**
 * Home container.
 * @module components/theme/NotFound/NotFound
 */

import React from 'react';

/**
 * Not found function.
 * @function NotFound
 * @returns {string} Markup of the not found page.
 */
const NotFound = () => (
  <div id="page-not-found">
    <h1>Doh! 404!</h1>
    <p>These are <em>not</em> the droids you are looking for!</p>
  </div>
);

export default NotFound;
