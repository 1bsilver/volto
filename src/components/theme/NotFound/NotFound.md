NotFound example source:

```jsx noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import BrowserRouter from 'react-router-dom/BrowserRouter';
import NotFound from './NotFound';
import StaticRouter from 'react-router-dom/StaticRouter';

const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  navigation: {
    items: [],
  },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <NotFound />
    </StaticRouter>
  </Provider>
</div>;
```
