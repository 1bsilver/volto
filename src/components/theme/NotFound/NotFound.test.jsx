import React from 'react';
import renderer from 'react-test-renderer';
import NotFound from './NotFound';

test('renders a not found component', () => {
  const component = renderer.create(
    <NotFound />
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
