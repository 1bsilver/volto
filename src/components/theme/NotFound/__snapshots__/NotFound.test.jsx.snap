// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a not found component 1`] = `
<div
  className="container"
>
  <h1>
    Doh! 404!
  </h1>
  <p>
    These are 
    <em>
      not
    </em>
     the droids you are looking for!
  </p>
</div>
`;
